src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.cs

```diff
@@ -1571,6 +1575,10 @@ private static void ReadTargetFrameworks(PackageSpec packageSpec, JsonTextReader
                         targetFrameworkInformation.RuntimeIdentifierGraphPath = jsonReader.ReadNextTokenAsString();
                         break;
 
+                    case "targetAlias":
+                        targetFrameworkInformation.TargetAlias = jsonReader.ReadNextTokenAsString();
```

Comment:
Does the restore code expect alias to always be set?

I think `TargetAlias` should be defaulted to `jsonReader.Value`'s value up on line 1517, and overridden here if the json specified it.

Reply:
Not really. 

This is an already processed packagespec (as in, if it needed an alias, it would've been generated and written).
We shouldn't make assumptions like that here.